### ISSUE TITLE
Seg fault in bipartitions code

### ISSUE THREAD

**Issue by james-d-mitchell** [gap-packages/Semigroups#608]
I get a seg fault when I do:

~~~~
test := function(S)
  local SS, XX, map1, map2, Y, T, R, z, r, xx;

  SS := DirectProduct(S, S);
  XX := GeneratorsOfSemigroup(SS);
  map1 := Embedding(SS, 1);
  map2 := Embedding(SS, 2);
  Y := List(GeneratorsOfSemigroup(S), x -> x ^ map1 * x ^ map2);
  T := Semigroup(Y);

  R := [RClassNC(T, One(SS))];

  for r in R do 
    for xx in XX do 
      z := xx * Representative(r);
      if not ForAny(R, C -> z in C) then 
        Add(R, RClassNC(T, z));
        Print("found ", Length(R), " so far\n");
      fi;
    od;
  od;
  return R;
end;
test(UniformBlockBijectionMonoid(6));
~~~~~

This is with GAP 4.10.1 and Semigroups `master` branch. It appears to intermittently seg fault in `BIPART_PERM_LEFT_QUO`, I haven't had the chance to investigate further at present.

**Comment by james-d-mitchell** [comment]
I think this may be a non-issue, I've rebuilt  everything, and can't seem to reproduce this. @flsmith also couldn't reproduce it.

**Comment by james-d-mitchell** [comment]
Now, I'm not sure, further investigation is required.

**Comment by james-d-mitchell** [comment]
This does indeed appear to be a bug.

**Comment by The-Beast** [comment]
I also got a seg fault.

**Comment by james-d-mitchell** [comment]
Resolved in v3.1.4
